src/api/c/scan.cpp

```diff
@@ -186,10 +186,10 @@ af_err af_scan_by_key(af_array *out, const af_array key, const af_array in, cons
         case s32:  res = scan_op<int    , int    >(key, in, dim, op, inclusive_scan); break;
         case u64:  res = scan_op<uintl  , uintl  >(key, in, dim, op, inclusive_scan); break;
         case s64:  res = scan_op<intl   , intl   >(key, in, dim, op, inclusive_scan); break;
-        case u16:  res = scan_op<ushort , uint   >(key, in, dim, op, inclusive_scan); break;
-        case s16:  res = scan_op<short  , int    >(key, in, dim, op, inclusive_scan); break;
-        case u8:   res = scan_op<uchar  , uint   >(key, in, dim, op, inclusive_scan); break;
-        case b8:   res = scan_op<char   , uint   >(key, in, dim, op, inclusive_scan); break;
+        case u16:  res = scan_op<uint   , uint   >(key, in, dim, op, inclusive_scan); break;
```

Comment:
I think it would be better to error out for these cases and make the user perform the cast themselves. That way no one is surprised.